doc/adr/0011-concept-exception-handling-backend.md

```diff
@@ -0,0 +1,33 @@
+# 10. How to use exceptions in the backend
+
+Date: 2024-06-28
+
+## Status
+
+Accepted
+
+## Context
+So far we used different approaches when to use Optionals, throw checked exceptions or unchecked exceptions. This also led
+to using exceptions from other layers. We want to streamline our approach.
+
+## Decision
+Layer-Specific Exceptions
+* By having each layer (ports, adapters, and application) define its own exceptions, we achieve a clear separation of concerns. This ensures that exceptions are relevant to the specific context in which they occur.
+* Avoiding a common exception package prevents mixing unrelated exceptions and promotes a more organized codebase.
```

Comment:
Maybe also add a short sentence what we had the idea of a common package for exceptions --> so that we can reuse exceptions in different layers